### ISSUE TITLE
Unable to use provider with `api_user_service_key`

### ISSUE THREAD

**Issue by mattmichal** [cloudflare/terraform-provider-cloudflare#1939]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.2.9
on linux_amd64

registry.terraform.io/cloudflare/cloudflare v3.24.0

### Affected resource(s)

- cloudflare_origin_ca_certificate

### Terraform configuration files

```hcl
# CLOUDFLARE_API_USER_SERVICE_KEY environment variable is given

terraform {
  required_version = "1.2.9"

  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "3.24"
    }
  }
}

provider "cloudflare" {}

data "cloudflare_ip_ranges" "ips" {}

output "cloudflare_ip_cidrs" {
  value = data.cloudflare_ip_ranges.ips.cidr_blocks
}
```

### Link to debug output

https://gist.github.com/mattmichal/7076354c10e4ae56c1524c4223a56ba7

### Expected output

Provider is configured and data calls can be executed.

### Actual output

```
** CLOUDFLARE_API_USER_SERVICE_KEY only
│ Error: error creating new Cloudflare client: invalid credentials: key & email must not be empty
│ 
│   with provider["registry.terraform.io/cloudflare/cloudflare"],
│   on providers.tf line 18, in provider "cloudflare":
│   18: provider "cloudflare" {}
```

### Steps to reproduce

1. Set CLOUDFLARE_API_USER_SERVICE_KEY environment variable
2. Add cloudflare provider to a project
3. Execute plan operation

### Additional factoids

My initial error was due to configuring both CLOUDFLARE_API_TOKEN and CLOUDFLARE_API_USER_SERVICE_KEY variables. An error is returned as expected due to the changes in #1907.

```
** CLOUDFLARE_API_TOKEN and CLOUDFLARE_API_USER_SERVICE_KEY
│ Error: Invalid combination of arguments
│ 
│   with provider["registry.terraform.io/cloudflare/cloudflare"],
│   on providers.tf line 13, in provider "cloudflare":
│   13: provider "cloudflare" {}
│ 
│ "api_token": only one of `api_key,api_token,api_user_service_key` can be
│ specified, but `api_token,api_user_service_key` were specified.
```

After seeing that PR and the 3.24.0 release notes, I removed the API token because the user service key is required in my production project due to references to `cloudflare_origin_ca_certificate` resources.

**Comment by jacobbednarz** [comment]
this has already been addressed in #1923 and slated for the next release. if you'd like to use it now, you can [build an unreleased version to use](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/using-non-released-versions.md).

**Comment by mattmichal** [comment]
I confirmed that #1919 describes the same issue. It seems like I searched open issues and not closed issues when looking for existing reports.
